### PR TITLE
docs: new document for Parallelize

### PR DESCRIPTION
**Pull request by rjha-splunk**
Adding new Documentation for Parallelize feature, it will help in single TCP traffic to decongest.

### CODE REVIEW

**Review by mstopa-splunk — APPROVED** [review]

**Review by ikheifets-splunk — APPROVED** [review]

File: docs/configuration.md [hunk 1]
```diff
@@ -333,3 +333,17 @@ ExecStart=/usr/bin/podman run \
 # Change your status port
 
 Use `SC4S_LISTEN_STATUS_PORT` to change the "status" port used by the internal health check process. The default value is `8080`.
+
+# Parallelize
+Similar to eBPF for TCP congestion of single heavy data stream we use parallelize feature.
+
```
Comment (jenworthington) [inline docs/configuration.md:339]: Use the parallelize feature to manage TCP congestion when using single heavy-data streams.

File: docs/configuration.md [hunk 2]
```diff
@@ -333,3 +333,17 @@ ExecStart=/usr/bin/podman run \
...
+This feature can be used with `SC4S_PARALLELIZE_NO_PARTITION`.
+
+| Variable | Values        | Description |
+|----------|---------------|-------------|
+| SC4S_ENABLE_PARALLELIZE=yes  | yes or no(default) | Use parallelize to leverage multithreading when consuming from a single TCP connection. |
```
Comment (jenworthington) [inline docs/configuration.md:344]: | Yes or No (default).

File: docs/configuration.md [hunk 3]
```diff
@@ -333,3 +333,17 @@ ExecStart=/usr/bin/podman run \
...
+|SC4S_PARALLELIZE_NO_PARTITION=4 | integer | Set number of threads to use,Default value is 4. |
```
Comment (jenworthington) [inline docs/configuration.md:345]: Integer | Set the number of threads to use, the default value is 4. |

File: docs/experiments.md [hunk 4]
```diff
@@ -8,6 +8,14 @@
 eBPF is a feature that leverages Linux kernel infrastructure to evenly distribute the load, especially in cases when there is a huge stream of messages incoming from a single appliance.
 To use the eBPF feature, you must have a host machine with and OS that supports eBPF. eBPF should be used only in cases when other ways of SC4S tuning fail. See the [instruction](./configuration.md#ebpf) for configuration details. 
 To learn more visit this [blog post.](https://www.syslog-ng.com/community/b/blog/posts/syslog-ng-4-2-extra-udp-performance)
+
+### Parallelize (TCP)
+Normally, SC4S processes incoming messages from a TCP connection in a single thread. While this works fine with many connections, it doesn't work efficiently when using a single or very few high-traffic connections. Using this feature allows SC4S to process log messages from a single high-traffic TCP connection in multiple threads, thus increasing processing performance on multi-core machines.
```
Comment (jenworthington) [inline docs/experiments.md:13]: SC4S processes incoming messages from a TCP connection in a single thread. While this is adequate for many connections, it doesn't work efficiently when using a single or few high-traffic connections. This feature allows SC4S to process log messages from a single high-traffic TCP connection in multiple threads, which increases processing performance on multi-core machines.

File: docs/experiments.md [hunk 5]
```diff
@@ -8,6 +8,14 @@
...
+
+See the [instruction](./configuration.md#parallelize)
```
Comment (jenworthington) [inline docs/experiments.md:15]: To learn more, see the [Configuration documentation](./configuration.md#parallelize) as well as this

File: docs/experiments.md [hunk 6]
```diff
@@ -8,6 +8,14 @@
...
+
+To learn more visit this [blog post.](https://www.syslog-ng.com/community/b/blog/posts/accelerating-single-tcp-connections-in-syslog-ng-parallelize)
```
Comment (jenworthington) [inline docs/experiments.md:17]: [blog post.](https://www.syslog-ng.com/community/b/blog/posts/accelerating-single-tcp-connections-in-syslog-ng-parallelize)

**Review by jenworthington — COMMENTED** [review]
reviewed for style and grammar

**Pull request merged**